src/views/Landing.vue

```diff
@@ -76,161 +124,274 @@
+                <span v-else>
+                    <v-card-text>Please login using 3Bot Connect or continue as guest.</v-card-text>
+                    <v-card-actions>
+                        <v-spacer></v-spacer>
+                        <v-btn id="guestLoginBtn" @click="guestLogin" text>Continue as Guest</v-btn>
+                        <v-btn
+                            id="threebotConnectLoginBtn"
+                            @click="threebotConnectLogin"
+                            text
+                        >Use 3Bot Connect</v-btn>
+                    </v-card-actions>
+                </span>
+            </v-card>
+            <GuestLogin v-else />
+        </v-dialog>
     </section>
 </template>
 <script>
-    import { mapActions, mapMutations, mapGetters } from 'vuex';
-    import { updateCurrentStream } from '@/utils/mediaDevicesUtils';
-    import { AvatarGenerator } from 'random-avatar-generator';
-    import Settings from '../components/Settings';
-    export default {
-        components: {
-            Settings,
-        },
-        data() {
-            return {
-                /* eslint-disable */
-                reg: new RegExp('(?:https://.*/room/)?(.*)'),
-                /* eslint-enable */
-                valid: false,
-                inviteUrlRules: [
-                    url => !!url || 'Invite url is required',
-                    url =>
-                        this.reg.test(url) || 'Invite url or room ID  invalid',
-                ],
-                inviteUrl: null,
-                myBackground: '',
-                showSettings: false,
-            };
-        },
-        async mounted() {
-          this.refreshMediaDevices().then(() => {
+import { mapActions, mapMutations, mapGetters } from 'vuex';
+import { updateCurrentStream } from '@/utils/mediaDevicesUtils';
+import { AvatarGenerator } from 'random-avatar-generator';
+import GuestLogin from '../components/GuestLogin'
+export default {
+    components: {
+        GuestLogin
+    },
+    data() {
+        return {
+            /* eslint-disable */
+            reg: new RegExp('(?:https://.*/room/)?(.*)'),
+            /* eslint-enable */
+            valid: false,
+            inviteUrlRules: [
+                url => !!url || 'Invite url is required',
+                url => this.reg.test(url) || 'Invite url or room ID  invalid',
+            ],
+            inviteUrl: null,
+            myBackground: '',
+            devices: [],
+            videoDevice: null,
+            audioDevice: null,
+            isLoginInAsGuest: false,
+            showLogin: false
+        };
+    },
+    mounted() {
+        if (this.$route.query && this.$route.query.redirect) {
+            this.showLogin = true
+        }
+        if (this.$route.query && this.$route.query.roomName) {
+            this.inviteUrl = this.$route.query.roomName;
+        }
+        navigator.mediaDevices.enumerateDevices().then(devices => {
```

Comment:
Why is this one changed to the navigator instead of the one in the mediaDeviceStore?

Reply:
Because I've merged it wrongly :(